controller/tap/server.go

```diff
@@ -30,23 +29,24 @@ const requireIDHeader = "l5d-require-id"
 const podIPIndex = "ip"
 const defaultMaxRps = 100.0
 
-type (
-	server struct {
-		tapPort             uint
-		k8sAPI              *k8s.API
-		controllerNamespace string
-	}
-)
+// GRPCTapServer describes the gRPC server implementing tap.Tap_TapByResourceServer
+type GRPCTapServer struct {
```

Comment:
I decided to make this public because of lint warnings when the type is returned outside the package in [main.go](https://github.com/linkerd/linkerd2/blob/f6362dfa805de9a009188014256ecd66e7dc3bfc/controller/cmd/tap/main.go#L49-L50)